docs/3.1/docs/canton/usermanual/performance.rst

```diff
@@ -506,69 +506,44 @@ If you submit a ledger API command (which can have multiple Daml commands), then
-         controller owner 
-         do 
-         return ()
-
-   template BatchFoo
-   with 
-      owner : Party 
-   where 
-      signatory owner 
-
-      choice Run : ()
-         with  
-         batch : [ContractId Example]
-         obs : Party
-         -- The observer here is a choice observer. Therefore, Canton will
-         -- ship a single view with the top node of this choice if the observer of the 
-         -- choice aligns with the observer of the contract.       
-         observer obs
-         controller owner 
-         do
-         forA_ batch (`exercise` Foo)
-
-
-As a rule, the number of views should depend on the number of stakeholder groups you have in your batch choice, 
-not the number of "batches" you process in parallel within one command.
+Currently, Canton creates a view for every action node in the transaction tree if the
+**participants that host their informees are not a subset of their parent view's informee participants**.
+A view's informee participants are all the participants that must be informed about the view.
+Therefore, to reduce the number of views, you should try to reduce either the number of times the set of informees grows
+or make sure most of your actions share the same pool of participants that must be informed. Alternatively, you can
+add informees to the action nodes close to the root (e.g., by including choice observers) so that their children nodes
+are aggregated in the initial view, since these children nodes will only target a subset of those participants.
+
+One concrete way to reduce the number of views being generated by a Daml model is to write batch commands and group the
+operations on the ledger based on their informees' participants and create a single choice and batch operations per
+informee's participants group.
+
+You can also reduce the number of views being generated for the children of a node (i.e., parent-child views) when
+creating your Daml model.
+With the below example as reference, instead of sending a batch with 20 exercise commands `Foo`,
+you can group your contracts by a set of stakeholders that cover the exercise's informee participants and send
+one exercise command on a generator contract with the list of contracts that in turn exercises the Foo choice
+per stakeholder group. This way you can perform these operations in a single view, instead of having them spawn multiple
+different views.
+
+For example, if the `doUpdate` bellow is called from a choice visible to the `owner` party alone, then whether the `efficient`
+flag is set or not will have a huge impact on the number of views created.
+In the choice `Foo` of the `Example` contract even though the participants to be informed are a subset of the
+contract's informee participants they don't get merged into a single view if this choice is called independently each time,
+since they produce multiple root actions each belonging to their own view. Finally, aligning the choice observer of `Run`
```

Comment:
```suggestion
In the choice `Foo` of the `Example` contract, even though the participants to be informed are a subset of the
contract's informee participants they are not merged into a single view if this choice is called independently each time,
since they produce multiple root actions each belonging to their own view. Finally, aligning the choice observer of `Run`
```
"their own view" is a little ambiguous here, on a quick scan it isn't clear if "they" refers to the informee participants (the previous 'they' of the sentence) or the root actions.